stdlib/http/src/main/ballerina/http/auth/authz_handler.bal

```diff
@@ -111,48 +111,39 @@ function checkForScopeMatch (string[] resourceScopes, string[] userScopes, strin
                                                                                                     returns boolean {
     boolean authorized = matchScopes(resourceScopes, userScopes);
     if (authorized) {
-        log:printDebug("Successfully authorized to access resource: " + resourceName + ", method: " +
-                method);
+        log:printDebug(function() returns string {
+            return "Successfully authorized to access resource: " + resourceName + ", method: " + method;
+        });
     } else {
-        log:printDebug("Authorization failure for resource: " + resourceName + ", method: " + method);
+        log:printDebug(function() returns string {
+            return"Authorization failure for resource: " + resourceName + ", method: " + method;
+        });
     }
     return authorized;
 }
 
 function HttpAuthzHandler::authorizeFromCache(string authzCacheKey) returns (boolean|()) {
-    match trap self.positiveAuthzCache {
-        cache:Cache cache => {
-            return check <boolean> cache.get(authzCacheKey);
-        }
-        error|() => {}
+    var positiveCache = trap self.positiveAuthzCache;
+    if (positiveCache is cache:Cache) {
+        return check <boolean> positiveCache.get(authzCacheKey);
     }
-    match trap self.negativeAuthzCache {
-        cache:Cache cache => {
-            return check <boolean> cache.get(authzCacheKey);
-        }
-        error|() => {}
+    var negativeCache =  trap self.negativeAuthzCache;
+    if (negativeCache is cache:Cache) {
+        return check <boolean> negativeCache.get(authzCacheKey);
```

Comment:
We need to change the method signature for return error if we are going to use check in here.

Reply:
Created issue[1] to track the requirement. We will attend to it after stabilizing the branch.
[1] https://github.com/ballerina-platform/ballerina-lang/issues/11284